src/tables.rs

```diff
@@ -13,6 +55,31 @@ pub fn commonly_mapped_to_nothing(c: char) -> bool {
     }
 }
 
+/// B.2 Mapping for case-folding used with NFKC.
+pub fn case_fold(s: &str) -> String {
+    let mut result = String::new();
+
+    // Each character either maps to a sequence of replacement characters,
+    // or is passed through as-is.
+    for c in s.chars() {
+        if let Some(entry) = table_lookup(RFC3454_B_2, c) {
```

Comment:
https://unicode-rs.github.io/unicode-normalization/unicode_normalization/trait.UnicodeNormalization.html#tymethod.nfkc this returns an iterator over the folded characters - it's used in the saslprep implementation already.